### PR TITLE
Only remove folders that contain no files recursively

### PR DESCRIPTION
**Pull request by Flamefire**
Otherwise 'include/libfoo/foo.h'  would be removed from `$CPATH`

Bug introduced in https://github.com/easybuilders/easybuild-framework/pull/3145

### CODE REVIEW

File: test/framework/filetools.py [hunk 1]
```diff
@@ -1724,6 +1724,32 @@ def test_search_file(self):
         for pattern in ['*foo', '(foo', ')foo', 'foo)', 'foo(']:
             self.assertErrorRegex(EasyBuildError, "Invalid search query", ft.search_file, [test_ecs], pattern)
 
+    def test_is_empty_folder(self):
+        tmpdir = tempfile.mkdtemp()
```
Comment (boegel) [inline test/framework/filetools.py]: @Flamefire There's a temporary directory unique to every test already available in `self.test_prefix`, you can use that one as a sandbox (it's also cleaned up automatically).

Comment (Flamefire) [reply test/framework/filetools.py]: Ok, thanks. Done and improved the readability of the test

File: easybuild/tools/filetools.py [hunk 2]
```diff
@@ -681,6 +681,14 @@ def search_file(paths, query, short=False, ignore_dirs=None, silent=False, filen
     return var_defs, hits
 
 
+def is_empty_folder(folder_path):
```
Comment (boegel) [inline easybuild/tools/filetools.py]: * I would prefer `directory` over `folder` (we don't use `folder` anywhere else)
* A directory containing an empty subdirectory is not really empty, so this should be renamed...
  How about `dir_contains_files` (and flipping the `True`/`False` too, of course)?

Comment (Flamefire) [reply easybuild/tools/filetools.py]: Renaming makes sense.

I was thinking about future extensions, e.g. adding parameters like `non_empty_if_contains_subdir` or so. But I didn't see it necessary, at least ATM. I still stand by the name: A directory containing only directories is still empty. But I see how your name avoids the ambiguity. Changing...

File: easybuild/tools/filetools.py [hunk 3]
```diff
@@ -681,6 +681,11 @@ def search_file(paths, query, short=False, ignore_dirs=None, silent=False, filen
     return var_defs, hits
 
 
+def dir_contains_files(folder_path):
+    """Return true if the given folder does not contain any files"""
```
Comment (boegel) [inline easybuild/tools/filetools.py]: @Flamefire Docstring needs fixing too

Comment (Flamefire) [reply easybuild/tools/filetools.py]: :+1:

**Pull request merged**